apps/backoffice-v2/src/pages/MerchantMonitoring/MerchantMonitoring.page.tsx

```diff
@@ -38,14 +42,16 @@ export const MerchantMonitoring: FunctionComponent = () => {
           </Link>
         )}
       </div>
+      <div className={`flex`}>
+        <Search value={search} onChange={onSearch} />
+      </div>
       <div className="flex flex-1 flex-col gap-6 overflow-auto">
-        {isNonEmptyArray(businessReports) && (
-          <MerchantMonitoringTable data={businessReports ?? []} />
-        )}
+        {isNonEmptyArray(businessReports) && <MerchantMonitoringTable data={businessReports} />}
         {Array.isArray(businessReports) && !businessReports.length && !isLoadingBusinessReports && (
           <NoBusinessReports />
         )}
         <div className={`mt-auto flex items-center gap-x-2`}>
+          <div className={`flex w-[12ch] text-sm`}>{`Page ${page} of ${totalPages}`}</div>
```

Comment:
Is the loading state covered or when the values are null/undefined? Could be worth adding `?? 0` or similar. Skeleton would be ideal.

Reply:
You're right. I thought it would be better to not present the entire pagination area before we get results (otherwise they are just a component that the user could click and fail), so I wrapped the entire `div` with a `!!totalPages` condition. Tell me what do you think about it

Reply:
I think these kind of layout shifts are jarring and I don't follow on why not having `totalPages` or them loading would fail the pagination?

Reply:
This is not a final design. I talked with Nitzan about him properly designing (UI + UX) the pagination components. This was a quick addition so that users could see that page they're in without reading the URL. The entire pagination `div` is not shown if we didn't get a response from the backend about the business reports. So, for a second while loading, it's not shown (which I think is better than "Page 0 of 0", it think it's a bit dumb as we don't even present the table or the empty view yet)